File: examples/openapi-sdk-petstore/kotlin/sdk/com/example/petstore/api/StoreApi.kt
package com.example.petstore.api

import com.example.petstore.models.*
import okhttp3.Response

public interface StoreApi {
    /** GET /store/inventory */
    suspend fun getInventory(
        options: RequestOptions
    ): Map<String, Int>

    /** GET /store/inventory */
    suspend fun getInventoryWithResponse(
        options: RequestOptions
    ): Pair<Map<String, Int>, Response>

    /** POST /store/order */
    suspend fun placeOrder(
        body: Order,
        options: RequestOptions
    ): Order

    /** POST /store/order */
    suspend fun placeOrderWithResponse(
        body: Order,
        options: RequestOptions
    ): Pair<Order, Response>

    /** GET /store/order/{orderId} */
    suspend fun getOrderById(
        orderId: Long,
        options: RequestOptions
    ): Order

    /** GET /store/order/{orderId} */
    suspend fun getOrderByIdWithResponse(
        orderId: Long,
        options: RequestOptions
    ): Pair<Order, Response>

    /** DELETE /store/order/{orderId} */
    suspend fun deleteOrder(
        orderId: Long,
        options: RequestOptions
    )

    /** DELETE /store/order/{orderId} */
    suspend fun deleteOrderWithResponse(
        orderId: Long,
        options: RequestOptions
    ): Response

    /** POST /shapes */
    suspend fun createShape(
        body: ByteArray,
        options: RequestOptions
    ): CreateShape_Response

    /** POST /shapes */
    suspend fun createShapeWithResponse(
        body: ByteArray,
        options: RequestOptions
    ): Pair<CreateShape_Response, Response>

    /** POST /measurements */
    suspend fun submitMeasurement(
        body: SubmitMeasurement_Body,
        options: RequestOptions
    )

    /** POST /measurements */
    suspend fun submitMeasurementWithResponse(
        body: SubmitMeasurement_Body,
        options: RequestOptions
    ): Response
}

File: examples/openapi-sdk-petstore/swift/sdk/API/StoreAPI.swift
import Foundation

public protocol StoreAPI {
    /// GET /store/inventory
    func getInventory(
        options: RequestOptions
    ) async throws -> [String: Int32]

    /// GET /store/inventory
    func getInventoryWithResponse(
        options: RequestOptions
    ) async throws -> ([String: Int32], HTTPURLResponse)

    /// POST /store/order
    func placeOrder(
        body: Order,
        options: RequestOptions
    ) async throws -> Order

    /// POST /store/order
    func placeOrderWithResponse(
        body: Order,
        options: RequestOptions
    ) async throws -> (Order, HTTPURLResponse)

    /// GET /store/order/{orderId}
    func getOrderById(
        orderId: Int64,
        options: RequestOptions
    ) async throws -> Order

    /// GET /store/order/{orderId}
    func getOrderByIdWithResponse(
        orderId: Int64,
        options: RequestOptions
    ) async throws -> (Order, HTTPURLResponse)

    /// DELETE /store/order/{orderId}
    func deleteOrder(
        orderId: Int64,
        options: RequestOptions
    ) async throws

    /// DELETE /store/order/{orderId}
    func deleteOrderWithResponse(
        orderId: Int64,
        options: RequestOptions
    ) async throws -> HTTPURLResponse

    /// POST /shapes
    func createShape(
        body: Data,
        options: RequestOptions
    ) async throws -> CreateShape_Response

    /// POST /shapes
    func createShapeWithResponse(
        body: Data,
        options: RequestOptions
    ) async throws -> (CreateShape_Response, HTTPURLResponse)

    /// POST /measurements
    func submitMeasurement(
        body: SubmitMeasurement_Body,
        options: RequestOptions
    ) async throws

    /// POST /measurements
    func submitMeasurementWithResponse(
        body: SubmitMeasurement_Body,
        options: RequestOptions
    ) async throws -> HTTPURLResponse
}
